pkg/cp/cmp/cmp.go

```diff
@@ -95,11 +114,11 @@ func IsEqualTree(o cp.Options, src, dst string) error {
 		return isEqualFile(src, dst)
 
 	case srcInfo.Mode()&os.ModeSymlink == os.ModeSymlink:
-		srcTarget, err := os.Readlink(src)
+		srcTarget, err := readLink(src)
```

Comment:
Another way to force a traversal of a file tree that does not exist is to use Walk:
https://pkg.go.dev/path/filepath#Walk
note that you can give it any function, and as long as that function returns reasonable results, it will all work. So have a function you pass to walk that returns things like os.FileInfo. 

Let me know if I'm off track. But keep in mind, the global variables thing is a sign that we need to attack the problem differently.